lua/neogit/lib/uv.lua

```diff
@@ -0,0 +1,22 @@
+local a = require 'plenary.async_lib'
+local async, await, uv = a.async, a.await, a.uv
+
+local M = {}
+
+M.read_file = async(function(path)
+  local err, fd = await(uv.fs_open(path, "r", 438))
+  if err then return err end
+
+  local err, stat = await(uv.fs_fstat(fd))
+  if err then return err end
+
+  local err, data = await(uv.fs_read(fd, stat.size, 0))
+  if err then return err end
+
+  local err = await(uv.fs_close(fd))
+  if err then return err end
+
+  return nil, data
+end)
+
```

Comment:
Are you no longer using the read_lines function? It had line splitting that supports windows line endings.

Reply:
https://github.com/TimUntersberger/neogit/blob/dc6ba8299f1826016b76aa673a111c8630ad08d0/lua/neogit/async/uv.lua#L24-L34
This one? 

That's only in #100 for now :D I can copy it over though.

Reply:
Ah sorry mb.